src/vm/customattribute.cpp

```diff
@@ -166,8 +166,15 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
         if (length == (ULONG)-1)
+        {
+            GCPROTECT_BEGIN(gc)
+            {   
+                if (arrayType == SERIALIZATION_TYPE_ENUM)
```

Comment:
Oh, and I do need you to sign that contribution license agreement (https://cla2.dotnetfoundation.org.). Our lawyers do require that :-)